src/components/modals/MigrateAccounts/steps/01-step-overview.js

```diff
@@ -0,0 +1,209 @@
+  margin-top: 15px;
+  & > * {
+    margin-bottom: 10px;
+  }
+`
+const Currency = styled.div``
+const Logo = styled.div`
+  margin-bottom: 15px;
+`
+const Title = styled.div`
+  margin-bottom: 10px;
+`
+const Footer = styled.div`
+  display: flex;
+  flex: 1;
+  color: ${p => p.theme.colors.wallet};
+  flex-direction: row;
+
+  > :nth-child(2) {
+    flex: 1;
+    margin-left: 14px;
+    margin-right: 14px;
+  }
+`
+
+const FooterDeviceSwitch = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  align-items: center;
+
+  > ${Text} {
+    margin-top: 21px;
+    &:hover {
+      text-decoration: underline;
+    }
+  }
+`
+
+const Exclamation = styled.div`
+  align-self: center;
+  width: 30px;
+  height: 30px;
+  border-radius: 30px;
+  background-color: ${p => p.theme.colors.pillActiveBackground};
+  align-items: center;
+  display: flex;
+  justify-content: center;
+`
+
+const NextDeviceWarning = styled.div`
+  display: flex;
+  background: ${p => p.theme.colors.pillActiveBackground};
+  border-radius: 4px;
+  padding: 14px 19px;
+  margin-bottom: 11px;
+  color: ${p => p.theme.colors.wallet};
+  align-items: center;
+  & > :first-child {
+    margin-right: 17px;
+  }
+  & > :nth-child(2) {
+    flex: 1;
+  }
+`
+
+const StepOverview = ({
+  migratableAccounts,
+  migratedAccounts,
+  totalMigratableAccounts,
+}: StepProps) => {
+  const totalMigratedAccounts = Object.keys(migratedAccounts).length
+  const migratableCurrencyIds = Object.keys(migratableAccounts)
+
+  return (
+    <Box align="center">
+      <TrackPage category="MigrateAccounts" name="Step1" />
+
+      <Logo>
+        <LedgerLiveLogo
+          width="58px"
+          height="58px"
+          icon={<img src={i('ledgerlive-logo.svg')} alt="" width={35} height={35} />}
+        />
+      </Logo>
+      <Title>
+        <Text ff="Museo Sans|Regular" fontSize={6} color="dark">
+          <Trans i18nKey="migrateAccounts.overview.title" />
+        </Text>
+      </Title>
+      <Text color="graphite" ff="Open Sans|Regular" fontSize={4}>
+        <Trans i18nKey="migrateAccounts.overview.description" />
+      </Text>
+
+      {migratableAccounts && (
+        <Wrapper>
+          {totalMigratedAccounts ? (
+            <NextDeviceWarning>
+              <IconExclamationCircle size={16} />
+              <Text ff="Open Sans|Bold" fontSize={4}>
+                <Trans
+                  i18nKey="migrateAccounts.overview.pendingDevices"
+                  values={{ totalMigratableAccounts }}
+                />
+              </Text>
+            </NextDeviceWarning>
+          ) : null}
+          {migratableCurrencyIds.map(currencyId => (
+            <Currency key={currencyId}>
+              <Text color="dark" ff="Open Sans|SemiBold" fontSize={4}>
+                <Trans
+                  i18nKey="migrateAccounts.overview.currency"
+                  values={{
+                    currency: migratableAccounts[currencyId][0].currency.name,
```

Comment:
in this map, could you make a local variable `migratableAccounts[currencyId]` to help visibility?
the `[0]` feels a bit type unsafe but i guess ok because we guarantee there is at least one item?

Reply:
`migratableAccounts` is an object that is made by grouping the migratable accounts, so if there are no accounts there is no group at that point.